### ISSUE TITLE
FastAPI testing for Mapping and Validation workflow. 

### ISSUE THREAD

**Issue by manjitapandey** [hotosm/tasking-manager#6617]
Go to the Explore Projects page by clicking[ here](https://tm-fastapi.naxa.com.np/explore).
**Testing the Explore Projects Page:**

- Verify that project cards are displayed across different pages.
- Confirm that the map next to each project card can be shown or hidden.
- Apply filters for specific projects; confirm that results update as expected.
- Download project data as a CSV file and confirm it downloads without issues.

**Testing Project Details Page:**

- Click on any project card to open its Project Details page.
- Check that you can both view and post project-specific questions and comments.
- Test marking projects as favourites by clicking on ‘mark as favourite’ on the bottom ribbon of the project details page. Also try removing the project from your favourite. 
- Confirm that similar projects and the contribution timeline are visible and accurate.

**Contributing to a Project:**

- Click on the Contribute button at the bottom right corner and confirm that it takes you to the project’s Tasks page.
- On the Tasks Page:
- Users should be able to view project’s instructions. 
- Users should be able to view the Contributions on the contribution tab. Try applying the filters of the contribution tab and verify the filtered data is correct.
- Verify that the status of tasks is visible and accurate when viewed on the map.
- Confirm that you can access the history (Activities and comments) of individual tasks. Also verify that you can post comments on the specific task.

**Mapping or Validating a Task:**
Select a task and click Map selected task. Verify that you are redirected to the iD Editor page as expected. Once on the iD Editor Page, please test the following mapping actions:

- Attempt to lock the task for mapping.
- Split the Task: Test splitting the task into smaller segments.
- Try marking the task as fully mapped.
- Mark the task as invalid (e.g., mark it as bad imagery )
- Test stopping the mapping process midway clicking on stop mapping.
- Switch Editors: Change the editor from iD Editor to JOSM and other available editors

Validation actions:

- Lock a single task for validation.
- Lock multiple mapped tasks for validation. 
- Test stopping the validation process midway clicking on stop validation.
- Try marking the tasks as both validated and invalidate and observe if the status is updated correctly on map through colours. 
- Try adding the comments on the task and confirm that its posted and tagged. 

Note: You can only map a single task at a time but validate more than one mapped task on one go.
Note: You don’t need to actually map or validate the tasks in order to mark them as mapped or validated. The projects here are only for testing purposes.

**Add the related issues on the comments**

**Comment by dusanzivkovic** [comment]
OSM username: dusziv
Date of testing: 14 Nov 2024
Issues I have encountered:

**Testing the Explore Projects Page:**

- Free text search only searches for text in the project description, not in the project title as well, as in regular TM.
- Downloaded CSV file contains only the data type (object), not the data.

**Testing Project Details Page:**

- I can't post a comment on the project details page. I get an error message.

**Mapping or Validating a Task:**

- I can't validate nor invalidate a task. I get an error message.
- However, if I add a comment to the task, (and then stop the validation), it remains recorded in the history tab.

**Comment by manjitapandey** [comment]
Thanks @dusanzivkovic , That's really helpful
Can you please share the project id where you encountered the error related project details and mapping and validating task. 
Also if you could share the related screenshot.

**Comment by dusanzivkovic** [comment]
@manjitapandey I can't post comment on any task I tried, here's a screenshot of my attempt with project 57.

![comment](https://github.com/user-attachments/assets/3e30fd57-edae-457a-82bf-b0593662fa77)

Then I tried to validate task 74 project 110. The task has previously been marked as ready for validation (light blue). I could not validate it (an error message briefly appeared in the bottom left). But now the color of the task has changed to white.

![task74](https://github.com/user-attachments/assets/59b355c5-a1c9-45eb-a7cc-e53003d53df7)

I have also put a comment on this task and I can see it in the task history.

![task74c](https://github.com/user-attachments/assets/6cbea735-d570-4ff3-ae0b-642719676b0a)

Possibly it is because the task status came with project, I can't see anyone in the contributors list

**Comment by adiatmad** [comment]
I can't post a comment on a specific task
![image](https://github.com/user-attachments/assets/09d4f15a-4f55-4d79-850a-7d53150e3724)

**Comment by adiatmad** [comment]
This step is FAILED on my end, using Firefox 132.0.2 (64-bit)

- Try marking the tasks as both validated and invalidate and observe if the status is updated correctly on map through colours. 

![image](https://github.com/user-attachments/assets/38e14ec7-3098-4214-897e-016ba7ecf9f5)

**Comment by manjitapandey** [comment]
Confirming the behavior on my device. Thanks for raising the issue. @dusanzivkovic , @adiatmad

**Comment by martien-176** [comment]
https://docs.google.com/document/d/1UaZQiePkFxmh26i8CvGmnehfTToc0EGUkceJNqvfFvY/edit?tab=t.0

This is a copy of the workflow doc with additional notes about test results.

I will write a recap of these hereafter.

**Comment by martien-176** [comment]
**Apply filters and use the search bar; verify that results update according to the filter/search criteria.**
- Button “More filters” does not turn red when using "More filter" -> “Interests”.
- Search works fine: both description and title is included
-  More filter -> Types of mapping: not working. Eg selecting "buildings exact match" shows all projects (21 total), including the ones without buildings

**Comment by manjitapandey** [comment]
@adiatmad , @dusanzivkovic, We have fixed the issues you raised. Can you confirm if the issues are resolved?

**Comment by dusanzivkovic** [comment]
@manjitapandey yes, I can confirm that issues with posting comments and validating tasks are now resolved, and the task status is correctly updated through colors.

**Comment by adiatmad** [comment]
> I can't post a comment on a specific task ![image](https://github.com/user-attachments/assets/09d4f15a-4f55-4d79-850a-7d53150e3724)

It's all fixed now, thanks a lot! @manjitapandey

**Comment by adiatmad** [comment]
> This step is FAILED on my end, using Firefox 132.0.2 (64-bit)
> 
>     * Try marking the tasks as both validated and invalidate and observe if the status is updated correctly on map through colours.
> 
> 
> ![image](https://github.com/user-attachments/assets/38e14ec7-3098-4214-897e-016ba7ecf9f5)

It's all fixed now, thanks a lot! @manjitapandey

**Comment by petya-kangalova** [comment]
Adding a few screenshots from the session. Understand that this is likely due to server load on testing site.
![Image](https://github.com/user-attachments/assets/4485db16-2874-462a-9358-a3542eccee06)


First blank and takes a few seconds to load
![Image](https://github.com/user-attachments/assets/d8caa266-507c-45c5-81a2-e9b7f934baa0)


Only see the grids:
![Image](https://github.com/user-attachments/assets/032af2f9-bdec-4f21-8f3b-764b7f1d3fa3)

**Comment by manjitapandey** [comment]
Hi all, the issue of not loading the basemap has been fixed.

**Comment by Pseudorandom-Pseudonym** [comment]
The "Projects to validate" filter does not function as desired. This is the case for both the FastAPI and the current HOT Tasking manager site. Please see the [issue](https://github.com/hotosm/tasking-manager/issues/6648) I published regarding this.

**Comment by martien-176** [comment]
https://tm-fastapi.naxa.com.np/explore

In the explore list view the author names are missing. See screenshot. 

![Image](https://github.com/user-attachments/assets/a7cfafa2-fecb-40aa-ac24-56cdb67b9ab2)
